packages/message/Collections.js

```diff
@@ -33,11 +34,14 @@ function constructCollectionId(name, version) {
  *
  * @alias module:Collections
  */
-const getCollectionIdFromMessage = (message) =>
-  constructCollectionId(
-    get(message, 'meta.collection.name'), get(message, 'meta.collection.version')
-  );
-
+const getCollectionIdFromMessage = (message) => {
+  const collectionName = get(message, 'meta.collection.name');
+  const collectionVersion = get(message, 'meta.collection.version');
+  if (!collectionName && !collectionVersion) {
+    throw new CumulusMessageError(`Collection name and version missing from message meta: ${JSON.stringify(message.meta)}`);
```

Comment:
Given that a Cumulus message without a collection is perfectly valid (I think), returning `undefined` here seems more appropriate.

Reply:
A Cumulus message with an *undefined* collection.... yes that's true, although if it is, having a granules or pdr payload will result in DB write failures.     I could go either way there, depending on our intention re: default message assumptions. 

ea55b31e2